### PR TITLE
Added -f option for input json file

### PR DESCRIPTION
**Pull request by tarzanek**
p99 percentile, max and output files
output to tagged files with buffer size and test name

### CODE REVIEW

**Comment by tarzanek** [comment]
and also graphs now show p95 as green dots and p99 as top of graph

File: diskplorer.py [hunk 1]
```diff
@@ -29,21 +29,24 @@
...
 (options, args) = optparser.parse_args()
 
 mountpoint = options.mountpoint
 filesize = options.filesize
 maxdepth = options.maxdepth
 buffer_size = options.buffer_size
-output_filename = options.output_filename
-raw_filename = options.raw_results_filename
 test_name = options.test_name
+output_filename = options.output_filename.format(buffer_size=buffer_size,test_name=test_name)
```
Comment (vladzcloudius) [inline diskplorer.py]: How would this work if I provided something like: `-o vladz_test`.

What I had in mind that string given with `-o` is then used for all output files: <name>.tmp, <name>.json, <name>.csv, <name>.svg.

This in particular make the `--raw-results` option not needed anymore.

**Review by vladzcloudius — CHANGES_REQUESTED** [review]
Please, split this patch.
You add much more than just a `-f` in it.
Let's not mix things up.

File: diskplorer.py [hunk 2]
```diff
@@ -166,8 +169,8 @@ def fix_y_axis(plt):
 plt.savefig(fname=output_filename)
 
 with open(raw_filename, 'w') as raw:
-    print('buffersize,concurrency,iops,lat_avg,lat_05,lat_95,lat_99,max', file=raw)
-    for concurrency, iops, lat_avg, lat_05, lat_95, lat_99,lat_max in zip(
-            concurrencies, iopses, latencies, latencies_05, latencies_95, latencies_99, latencies_max):
-        print('{buffer_size},{concurrency},{iops},{lat_avg},{lat_05},{lat_95},{lat_99},{lat_max}'
+    print('buffersize,concurrency,iops,lat_avg(us),lat_05(us),lat_95(us),lat_99(us),lat_9999(us),lat_max(us)', file=raw)
```
Comment (vladzcloudius) [inline diskplorer.py:169]: Latencies are in `ns`. In order to print them in `us` you need to divide them by 1000.

**Review by vladzcloudius — CHANGES_REQUESTED** [review]
You also add max.
You are still mixing at least 3 things in the same patch:
  * Adding new latency points.
  * switching to use ints instead of float.
  * Printing latencies in `us` instead of `ns`.

 If you split them in separate patches it would not only make easier not only reviewing them but also addressing my comments. ;)

**Comment by tarzanek** [comment]
split into 3 patches now, issues addressed

**Review by vladzcloudius — APPROVED** [review]
LGTM

**Review by vladzcloudius — APPROVED** [review]

File: diskplorer.py [hunk 3]
```diff
@@ -119,19 +119,28 @@ def run_job():
...
+    latency_99 = float(job[stat_label]['clat_ns']['percentile']['99.000000'])
+    latency_9999 = float(job[stat_label]['clat_ns']['percentile']['99.990000'])
+    latency_max = float(job[stat_label]['clat_ns']['max'])
     latency_stddev = float(job[stat_label]['clat_ns']['stddev'])
     iops = float(job[stat_label]['iops'])
     concurrencies.append(concurrency)
-    latencies.append(latency)
-    latencies_05.append(latency_05)
-    latencies_95.append(latency_95)
+    latencies.append(latency/1000)
```
Comment (vladzcloudius) [inline diskplorer.py:138]: You also switch to print latencies in `us` instead of `ns`. 
Either split or mention this in the patch description.

File: diskplorer.py [hunk 4]
```diff
@@ -148,17 +157,18 @@ def fix_y_axis(plt):
 
 ax2 = ax1.twinx()
 #ax2.plot(concurrencies, latencies, 'r-+')
-ax2.errorbar(concurrencies, latencies, yerr=[latencies_05, latencies_95], color='r')
-ax2.set_ylabel(u'average latency (ns)', color='r')
+ax2.errorbar(concurrencies, latencies, yerr=[latencies_05, latencies_99], color='r')
```
Comment (vladzcloudius) [inline diskplorer.py]: Why did you need to switch from 05-95 to 05-99.
Let's keep the original 05-95th as min-max by default and we can introduce a parameter that would allow choosing from other options (enum) for both left and right sides of the range.

**Review by vladzcloudius — CHANGES_REQUESTED** [review]

File: diskplorer.py (hunk 3 above)
Comment (tarzanek) [reply diskplorer.py:138]: last patch says that:
" Print latencies in us, add 99p, 9999p and max latencies "

File: diskplorer.py (hunk 4 above)
Comment (tarzanek) [reply diskplorer.py]: creating a commit that reverts to 95p

**Comment by PeterDaveHello** [comment]
@tarzanek would you like to send the pull request to upstream: https://github.com/avikivity/diskplorer?

**Comment by tarzanek** [comment]
> @tarzanek would you like to send the pull request to upstream: https://github.com/avikivity/diskplorer?

@PeterDaveHello feel free to do it on my behalf, I won't get to it anytime soon
